src/unitxt/metrics.py

```diff
@@ -70,26 +91,40 @@ def process(self, stream: Stream, stream_name: str = None) -> Generator:
 
             instance["score"]["instance"].update(instance_score)
 
-            references.append(refs)
-            predictions.append(pred)
-            instances.append(instance)
-
-        result = self._compute(references, predictions)
+        result = self._compute(references, predictions, additional_inputs if self.use_additional_inputs else None)
 
         global_score.update(result)
 
         for instance in instances:
             instance["score"]["global"] = global_score
             yield instance
 
-    def _compute(self, references: List[List[str]], predictions: List[str]) -> dict:
-        result = self.compute(references, predictions)
+    def _compute(self, references: List[List[str]], predictions: List[str], additional_inputs: List[Any]) -> dict:
+        result = self.compute_with_additional_inputs(references, predictions, additional_inputs)
         result["score"] = result[self.main_score]
         result["score_name"] = self.main_score
         return result
 
     @abstractmethod
-    def compute(self, references: List[List[str]], predictions: List[str]) -> dict:
+    def compute_with_additional_inputs(
```

Comment:
why was this function renamed? given that there is not regular `compute`, why not just keep it `compute`?